.circleci/release_ios.yml

```diff
@@ -0,0 +1,243 @@
+          shell: /bin/bash -l
+          environment:
+            TMP_WORKSPACE: *tmp-workspace
+            ENV_VARS_FILE: *env-vars-file
+
+  generate-framework:
+    executor: mac-os-executor
+
+    steps:
+      - checkout
+      - run:
+          name: Create Framework
+          command: ./gradlew shared:assembleTwilioPasskeysAuthenticationXCFramework
+      - run:
+          name: Compress Artifacts
+          command: |
+            ln -s shared/build/XCFrameworks/debug/TwilioPasskeysAuthentication.xcframework TwilioPasskeysAuthentication.xcframework
+            zip -r debugFramework.zip TwilioPasskeysAuthentication.xcframework
+            rm TwilioPasskeysAuthentication.xcframework
+            ln -s shared/build/XCFrameworks/release/TwilioPasskeysAuthentication.xcframework TwilioPasskeysAuthentication.xcframework
+            zip -r releaseFramework.zip TwilioPasskeysAuthentication.xcframework
+            rm TwilioPasskeysAuthentication.xcframework
+      - store_artifacts:
+          path: debugFramework.zip
+      - store_artifacts:
+          path: releaseFramework.zip
+      - persist_to_workspace:
+          root: *workspace
+          paths:
+            - debugFramework.zip
+            - releaseFramework.zip
+
+  generate-ios-simulator-app-using-internal-sdk:
+    executor: mac-os-executor
+
+    steps:
+      - checkout
+      - ruby/install-deps
+      - attach_workspace:
+          at: *workspace
+      - run:
+          name: Install Gem dependencies
+          command: |
+            cd iosApp
+            bundle install
+      - set-sample-backend-for-ios
+      - setup-certificates
+      - run:
+          name: Generate simulator build
+          command: |
+            cd iosApp
+            bundle exec fastlane simulator_app
+            mv simulator_build/Build/Products/Debug-iphonesimulator/iosApp.app sample-app.app
+            zip -r sample-app.zip sample-app.app
+            export IOS_SIMULATOR_APP_URL="https://output.circle-artifacts.com/output/job/${CIRCLE_WORKFLOW_JOB_ID}/artifacts/${CIRCLE_NODE_INDEX}/iosApp/sample-app.zip"
+            echo $IOS_SIMULATOR_APP_URL
+            cd ..
+            ./scripts/add_env_variable_to_file.sh IOS_SIMULATOR_APP_URL $IOS_SIMULATOR_APP_URL $TMP_WORKSPACE $ENV_VARS_FILE
+          environment:
+            TMP_WORKSPACE: *tmp-workspace
+            ENV_VARS_FILE: *env-vars-file
+      - cleanup-signing
+      - store_artifacts:
+          path: iosApp/sample-app.zip
+      - persist_to_workspace:
+          root: *workspace
+          paths:
+            - *tmp-workspace
+
+  run-simulator-ios-internal-sdk-e2e-tests:
+    executor: ubuntu-small-executor
+
+    parameters:
+      waiting_job_name:
+        default: ""
+        type: string
+    steps:
+      - checkout
+      - attach_workspace:
+          at: *workspace
+      - run:
+          name: Start automation pipeline to run E2E tests
+          command: |
+            source "$TMP_WORKSPACE/$ENV_VARS_FILE"
```

Comment:
Can this be unified with: https://github.com/twilio/twilio-verify-passkeys/blob/ee8b1292751aa2843768bf4786814d5242c45e28/.circleci/config.yml#L545 ?
It could be a script or it can be a common step in circleci, using `circleci config pack`